engines/dfc_provider/app/services/dfc_provider/authorization_control.rb

```diff
@@ -14,10 +14,10 @@ def process
     end
 
     def decode_token
+      rsa_public = OpenSSL::PKey::RSA.new(DFC_PUBLIC_KEY)
       data = JWT.decode(
         @access_token,
-        nil,
-        false
+        rsa_public, true, { algorithm: 'RS256' }
```

Comment:
This looks good to test the signature, but I am wondering what will be the method output if it fails? Maybe an error is thrown? Would be good to add tests for this.

Reply:
I added some basic tests in e9c1bf3eecefe164f3e148fad1182da8705c35ed to get us started... I have it just rescue the error and return `nil` for now but we could definitely enhance it if we need to.